src/link.c

```diff
@@ -0,0 +1,2259 @@
+    FillWindowPixelBuffer(2, PIXEL_FILL(0));
+    AddTextPrinterParameterized3(1, 3, 2, 0, sLinkErrorTextColor, 0, gText_CommErrorCheckConnections);
+    PutWindowTilemap(1);
+    PutWindowTilemap(2);
+    CopyWindowToVram(1, 0);
+    CopyWindowToVram(2, 3);
+    ShowBg(0);
+}
+
+static void CB2_PrintErrorMessage(void)
+{
+    switch (gMain.state)
+    {
+    case  00:
+        if (sLinkErrorBuffer.unk_06)
+        {
+            sub_800AE1C();
+        }
+        else
+        {
+            sub_800AED0();
+        }
+        break;
+    case  30:
+        PlaySE(SE_BOO);
+        break;
+    case  60:
+        PlaySE(SE_BOO);
+        break;
+    case  90:
+        PlaySE(SE_BOO);
+        break;
+    case 130:
+        if (gWirelessCommType == 2)
+        {
+            AddTextPrinterParameterized3(0, 3, 2, 20, sLinkErrorTextColor, 0, gText_ABtnTitleScreen);
+        }
+        else if (gWirelessCommType == 1)
+        {
+            AddTextPrinterParameterized3(0, 3, 2, 20, sLinkErrorTextColor, 0, gText_ABtnRegistrationCounter);
+        }
+        break;
+    }
+    if (gMain.state == 160)
+    {
+        if (gWirelessCommType == 1)
+        {
+            if (JOY_NEW(A_BUTTON))
+            {
+                sub_812B484();
+                PlaySE(SE_PIN);
+                gWirelessCommType = 0;
+                sLinkErrorBuffer.unk_06 = 0;
+                sub_8079B7C();
+            }
+        }
+        else if (gWirelessCommType == 2)
+        {
+            if (JOY_NEW(A_BUTTON))
+            {
+                sub_812B484();
+                rfu_REQ_stopMode();
+                rfu_waitREQComplete();
+                DoSoftReset();
+            }
+        }
+    }
+    if (gMain.state != 160)
+    {
+        gMain.state++;
+    }
+}
+
+bool8 GetSioMultiSI(void)
+{
+    return (REG_SIOCNT & 0x04) != 0;
+}
+
+static bool8 IsSioMultiMaster(void)
+{
+    return (REG_SIOCNT & 0x8) && !(REG_SIOCNT & 0x04);
+}
+
+bool8 IsLinkConnectionEstablished(void)
+{
+    return EXTRACT_CONN_ESTABLISHED(gLinkStatus);
+}
+
+void SetSuppressLinkErrorMessage(bool8 flag)
+{
+    gSuppressLinkErrorMessage = flag;
+}
+
+bool8 HasLinkErrorOccurred(void)
+{
+    return gLinkErrorOccurred;
+}
+
+void sub_800B0B4(void)
+{
+    struct LinkPlayerBlock *block;
+
+    InitLocalLinkPlayer();
+    block = &gLocalLinkPlayerBlock;
+    block->linkPlayer = gLocalLinkPlayer;
+    memcpy(block->magic1, sASCIIGameFreakInc, sizeof(block->magic1) - 1);
+    memcpy(block->magic2, sASCIIGameFreakInc, sizeof(block->magic2) - 1);
+    memcpy(gBlockSendBuffer, block, sizeof(*block));
+}
+
+void sub_800B110(u32 who)
+{
+    u8 who_ = who;
+    struct LinkPlayerBlock *block;
+    struct LinkPlayer *player;
+
+    block = (struct LinkPlayerBlock *)gBlockRecvBuffer[who_];
+    player = &gLinkPlayers[who_];
+    *player = block->linkPlayer;
+    sub_800B284(player);
+    if (strcmp(block->magic1, sASCIIGameFreakInc) != 0 || strcmp(block->magic2, sASCIIGameFreakInc) != 0)
```

Comment:
Is it good to write `!= 0` explicitly?

Reply:
For strcmp and memcmp it's highly recommended